crates/chia-tools/src/bin/benchmark-condition-costs.rs

```diff
@@ -0,0 +1,534 @@
+                &mut ret,
+                &mut state,
+                parent_id,
+                puz_hash_node_ptr,
+                hundred,
+                conditions,
+                flags,
+                &mut cost,
+                &TEST_CONSTANTS,
+            )
+            .expect("process_single_spend");
+
+            MempoolVisitor::post_process(&allocator, &state, &mut ret).expect("post_process");
+            validate_conditions(&allocator, &ret, &state, spends, flags)
+                .expect("validate_conditions");
+            validate_signature(&state, &signature, flags, None).expect("validate_signature");
+
+            let elapsed = start.elapsed();
+            // the first run is a warmup
+            if i > 1 {
+                samples.push(((i * multiplier) as f64, elapsed.as_nanos() as f64));
+            }
+            // add costs to tally
+            total_cost += ret.cost;
+            total_count += 1;
+
+            // make the conditions list longer
+            conditions = if matches!(
+                cond.opcode,
+                opcodes::SEND_MESSAGE
+                    | opcodes::ASSERT_PUZZLE_ANNOUNCEMENT
+                    | opcodes::ASSERT_COIN_ANNOUNCEMENT
+            ) {
+                cons_two_conditions(&mut allocator, conditions).expect("cons_condition")
+            } else {
+                cons_condition(&mut allocator, conditions).expect("cons_condition")
+            };
+        }
+        // reset allocator before next condition test
+        let (slope, offset): (f64, f64) = linear_regression_of(&samples).expect("linreg failed");
+        if cond.cost > 0 {
+            let cost_per_ns = cond.cost as f64 / slope;
+            cost_factors.push(cost_per_ns);
+            println!(
+                "condition: {} slope: {slope} cost-per-nanosecond: {cost_per_ns}",
+                cond.opcode
+            );
+        } else {
+            let cost_per_ns = cost_factors.iter().sum::<f64>() / cost_factors.len() as f64;
+            println!(
+                "condition: {} slope: {slope} computed-cost: {}",
+                cond.opcode,
+                slope * cost_per_ns
+            );
+        };
+        let mut plot = Figure::new();
+        plot.axes2d()
+            .set_x_label("number of conditions", &[])
+            .set_y_label("nanoseconds", &[])
+            .points(
+                samples.iter().map(|v| v.0),
+                samples.iter().map(|v| v.1),
+                &[PlotOption::PointSymbol('o')],
+            )
+            .lines(
+                [0, REPS * multiplier],
+                [offset, offset + slope * ((REPS * multiplier) as f64)],
+                &[],
+            );
+        plot.save_to_png(format!("condition-{}.png", cond.opcode), 1024, 768)
+            .expect("save svg");
+        plot.close();
+        allocator.restore_checkpoint(&cp);
+    }
+
+    println!("Total Cost: {total_cost}");
+    println!("Total Count: {total_count}");
```

Comment:
I think you can removed `total_count` too

Reply:
Done